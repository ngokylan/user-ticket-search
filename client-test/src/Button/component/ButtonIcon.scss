@import '../../_lib/style/index';
.btn {
  &__icon {
    margin-left: 12px;
    margin-right: 12px;
  }
}
