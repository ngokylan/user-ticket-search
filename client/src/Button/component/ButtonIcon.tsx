import React, { ReactNode } from 'react';

export type ButtonIconPropsType = {
  icon?: ReactNode;
  id?: string;
};

function ButtonIcon({ icon, id }: ButtonIconPropsType) {
  return (
    <span
      className="btn__icon"
      data-testid={`btn-icon-${id || 'default'}`}
    >
      {icon}
    </span>
  );
}

export default ButtonIcon;
