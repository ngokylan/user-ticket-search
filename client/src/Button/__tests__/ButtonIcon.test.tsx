import React from 'react';
import { mount } from 'enzyme';
import ButtonIcon, { ButtonIconPropsType } from '../component/ButtonIcon';

// Expected default props.
const defaultProps: ButtonIconPropsType = undefined;
const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = mount(<ButtonIcon {...props} />);
  const component = wrapper.find('.btn__icon');

  return { props, wrapper, component };
};

describe('Button::ButtonIcon', () => {
  describe('rendering', () => {
    it('should render component without any crash', () => {
      const { wrapper } = mountComponent();
      expect(wrapper).toHaveLength(1);
    });

    it.each([[{}], [{ id: 'custom-id', icon: 'custom-icon' }]])(
      'should be match with snapshot',
      props => {
        const { wrapper } = mountComponent(props);
        expect(wrapper).toMatchSnapshot();
      }
    );
  });

  describe('props', () => {
    describe('default', () => {
      it('should have expected default props', () => {
        expect(JSON.stringify(defaultProps)).toBe(
          JSON.stringify(ButtonIcon.defaultProps)
        );
      });

      it.each([
        ['btn-icon-default', {}],
        ['btn-icon-custom-id', { id: 'custom-id' }],
      ])('should render test ID properly', (expectedTestId, props) => {
        const { component } = mountComponent(props);
        expect(component.prop('data-testid')).toBe(expectedTestId);
      });
    });

    describe('children', () => {
      it('should render children', () => {
        const { wrapper } = mountComponent({ icon: 'icon' });
        expect(wrapper.find('span').text()).toBe('icon');
      });
    });
  });
});
