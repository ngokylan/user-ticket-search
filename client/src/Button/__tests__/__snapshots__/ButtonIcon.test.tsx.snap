// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button::ButtonIcon rendering should be match with snapshot 1`] = `
<span
  className="btn__icon"
  data-testid="btn-icon-default"
/>
`;

exports[`Button::ButtonIcon rendering should be match with snapshot 2`] = `
<span
  className="btn__icon"
  data-testid="btn-icon-custom-id"
>
  custom-icon
</span>
`;
